azure-iot-device/tests/common/pipeline/pipeline_stage_test.py

```diff
@@ -70,10 +76,10 @@ def add_unknown_ops_tests(cls, module, all_ops, handled_ops):
     unknown_ops = all_except(all_items=all_ops, items_to_exclude=handled_ops)
 
     @pytest.mark.describe("{} - .run_op() -- unknown and unhandled operations".format(cls.__name__))
-    class LocalTestObject(StageTestBase):
-        @pytest.fixture
-        def op(self, op_cls, mocker):
-            op = make_mock_op_or_event(op_cls)
+    class LocalTestObject(StageRunOpTestBase):
```

Comment:
>StageRunOpTestBase [](start = 26, length = 18)

You may know this, but this is a hidden gotcha with this style of testing that you might not have seen before.  StageRunOpBaseTests tests will run only with ops in the unkonwn_op array because that's how the op fixture is defined in this class.  I think the coverage is OK for now, better than it was before, but it's something to keep in mind.

Reply:
Yup. That's why in the next step of the revisions StageRunOpTestBAse, this base class needs to be extended to be used across all other .run_op() tests in all testing modules.

---
In reply to: [338069969](https://github.com/Azure/azure-iot-sdk-python/pull/369#discussion_r338069969) [](ancestors = 338069969)